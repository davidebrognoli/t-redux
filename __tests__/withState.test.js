import React from 'react'
import renderer from 'react-test-renderer'
import TestComponent from './fake/TestComponent'
import withState from '../lib/withState'
import dispatcher from '../lib/dispatcher'

test('withState should wrap', () => {
  const component = renderer.create(<TestComponent />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('withState supports partial application ', () => {
  const A = ({ name }) => <div>Hello, {name}!</div>
  const B = ({ name }) => <div>Ciao, {name}!</div>

  const withName = withState([], {name: 'Dan'})

  const AWithName = withName(A)
  const BWithName = withName(B)

  let tree = renderer.create(<AWithName />).toJSON()
  expect(tree).toMatchSnapshot()

  tree = renderer.create(<BWithName />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('withState should accept a pre-middleware function', () => {
  const A = ({ name }) => <div>Hello, {name}!</div>
  let done = false
  const middleware = (state, action) => { done = true }

  const Wrapped = withState([], {}, middleware)(A)
  const component = renderer.create(<Wrapped />)
  let tree = component.toJSON()

  dispatcher.dispatch({type: 'FAKE_ACTION', content: {}})

  expect(tree).toMatchSnapshot()
  expect(done).toBeTruthy()
})
