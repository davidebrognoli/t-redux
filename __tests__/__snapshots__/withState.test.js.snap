// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withState should accept a pre-middleware function 1`] = `
<div>
  Hello, 
  !
</div>
`;

exports[`withState should wrap 1`] = `
<div>
  Hello
</div>
`;

exports[`withState supports partial application  1`] = `
<div>
  Hello, 
  Dan
  !
</div>
`;

exports[`withState supports partial application  2`] = `
<div>
  Ciao, 
  Dan
  !
</div>
`;
